packages/libs/client-utils/src/core/yaml-converter.ts

```diff
@@ -0,0 +1,233 @@
+import type {
+  ChainId,
+  IExecutionPayloadObject,
+  IPactCommand,
+} from '@kadena/client';
+import { readFileSync } from 'fs';
+import yaml from 'js-yaml';
+import { join } from 'path';
+
+interface IParseContext {
+  cwd: string;
+  tplPath: string;
+  tplString: PartsAndHoles;
+}
+
+interface IParseContextStepTwo extends IParseContext {
+  filledYamlString: FilledYamlString;
+}
+interface IParseContextStepThree extends IParseContextStepTwo {
+  tplTx: Omit<IKdaToolTransaction, 'codeFile'> & { code?: string };
+}
+
+interface IKdaToolTransaction {
+  codeFile?: string;
+  command: string;
+  meta: {
+    chainId: string;
+    sender: string;
+    gasLimit: number;
+    gasPrice: number;
+    ttl: number;
+  };
+  networkId: string;
+  data: Record<string, any>;
+  signers: Array<{ public: string }>;
+  nonce: string;
+}
+
+interface TplHoleTriple {
+  literal: string;
+}
+interface TplHoleDouble {
+  parsed: string;
+}
+type TplHole = TplHoleTriple | TplHoleDouble;
+type TplPart = string;
+type PartsAndHoles = [TplPart[], TplHole[]];
+type GetPartsAndHoles = (path: string, cwd?: string) => IParseContext;
+type FilledYamlString = string;
+
+type ReplaceHoles = (
+  ctx: IParseContext,
+  holes: Record<string, string | number>,
+) => IParseContextStepTwo;
+
+type ParseYamlKdaTx = (
+  ctx: IParseContextStepTwo,
+  args: Record<string, string | number>,
+) => IParseContextStepThree;
+
+type ConvertKdaToolToKadenaClientTx = (
+  kdaToolTx: IParseContextStepThree['tplTx'],
+) => IPactCommand;
+
+type ConvertYamlToKadenaClientTx = (
+  path: string,
+  args: Record<string, string | number>,
+  cwd?: string,
+) => IPactCommand;
+
+// Responsinble for splitting a string into parts and holes
+export const getPartsAndHoles = (text: string) => {
+  return text.split(/{{(.*?)}}}?/g).reduce(
+    (acc, curr, i) => {
+      // if i is even, then it's a part
+      // if i is odd, then it's a hole
+      if (i % 2 === 0) {
+        acc[0].push(curr);
+      } else {
+        if (curr.startsWith('{')) {
+          acc[1].push({ literal: curr.slice(1) });
+          return acc;
+        }
+        acc[1].push({ literal: curr });
+      }
+      return acc;
+    },
+    [[], []] as PartsAndHoles,
+  );
+};
+
+// Responsible for reading a file and returning a context with parts and holes
+export const getPartsAndHolesInCtx: GetPartsAndHoles = (
+  path,
+  cwd = process.cwd(),
+) => {
+  const file = readFileSync(join(cwd, path), 'utf-8').toString();
```

Comment:
client utils should be executable on all environment not only nodejs.
if this function accepts the actual content instead of the file path then you don't need to use readFileSync.

Reply:
We need the path as the templateFiles can contain `codeFile` argument that's relative to the template file. Maybe we should move the file-specific functions to `@kadena/client-utils-node`? Or make `@kadena/client-utils/nodejs`?

Reply:
we can export them on `@kadena/client-utils/nodejs`